editors/code/src/main.ts

```diff
@@ -156,16 +156,19 @@ export async function deactivate() {
 async function bootstrap(config: Config, state: PersistentState): Promise<string> {
     await fs.mkdir(config.globalStoragePath, { recursive: true });
 
+    if (config.package.releaseTag !== NIGHTLY_TAG) {
+        await state.updateNightlyReleaseId(undefined);
+    }
     await bootstrapExtension(config, state);
     const path = await bootstrapServer(config, state);
-
     return path;
 }
 
 async function bootstrapExtension(config: Config, state: PersistentState): Promise<void> {
     if (config.package.releaseTag === null) return;
+    const currentExtensionIsNightly = config.package.releaseTag === NIGHTLY_TAG;
```

Comment:
We have one more check on line 159. Letss add `config.isCurrentExtensionNightly` or some such.